lib/utils.js

```diff
@@ -148,84 +154,96 @@ export function getDefaultAccount() {
-  constructor(contract) {
-    this.contract = contract;
-    for (const i in this.contract) {
-      if (this[i] === undefined){
-        this[i] = this.contract[i];
+export const ExtendTruffleContract = superclass =>
+  class {
+    constructor(contract) {
+      this.contract = contract;
+      for (const i in this.contract) {
+        if (this[i] === undefined) {
+          this[i] = this.contract[i];
+        }
       }
+      // for (var prop in this.contract) {
+      //   if (!this.hasOwnProperty(prop)) {
+      //     this[prop] = superclass[prop];
+      //   }
+      // }
     }
-    // for (var prop in this.contract) {
-    //   if (!this.hasOwnProperty(prop)) {
-    //     this[prop] = superclass[prop];
-    //   }
-    // }
-  }
 
-  static async new() {
-    return superclass.new()
-      .then((contract) => {
-        return new this(contract);
-      }, (ex) => { throw ex; });
-  }
+    static async new() {
+      return superclass.new().then(
+        contract => {
+          return new this(contract);
+        },
+        ex => {
+          throw ex;
+        }
+      );
+    }
 
-  static async at(address) {
-    return superclass.at(address)
-      .then((contract) => {
-        return new this(contract);
-      }, (ex) => { throw ex; });
-  }
+    static async at(address) {
+      return superclass.at(address).then(
+        contract => {
+          return new this(contract);
+        },
+        ex => {
+          throw ex;
+        }
+      );
+    }
 
-  static async deployed() {
-    return superclass.deployed()
-      .then((contract) => {
-        return new this(contract);
-      }, (ex) => { throw ex; });
-  }
+    static async deployed() {
+      return superclass.deployed().then(
+        contract => {
+          return new this(contract);
+        },
+        ex => {
+          throw ex;
+        }
+      );
+    }
 
-  /**
-   * Call setParameters on this scheme.
-   * Returns promise of parameters hash.
-   * If there are any parameters, then this function must be overridden by the subclass to provide them.
-   * @param overrides -- object with properties whose names are expected by the scheme to correspond to parameters.  Overrides the defaults.
-   *
-   * Should have the following properties:
-   *
-   *  for all:
-   *    voteParametersHash
-   *    votingMachine -- address
-   *
-   *  for ContributionReward:
-   *    orgNativeTokenFee -- number
-   *    schemeNativeTokenFee -- number
-   */
-  // eslint-disable-next-line no-unused-vars
-  async setParams(params) {
-    return await '';
-  }
+    /**
+     * Call setParameters on this scheme.
+     * Returns promise of parameters hash.
+     * If there are any parameters, then this function must be overridden by the subclass to provide them.
+     * @param overrides -- object with properties whose names are expected by the scheme to correspond to parameters.  Overrides the defaults.
+     *
+     * Should have the following properties:
```

Comment:
> Should have the following properties:
 -   *		 +     *
 -   *  for all:		 +     *  for all:
 -   *    voteParametersHash		 +     *    voteParametersHash
 -   *    votingMachine -- address		 +     *    votingMachine -- address
 -   *		 +     *
 -   *  for ContributionReward:		 +     *  for ContributionReward:
 -   *    orgNativeTokenFee -- number		 +     *    orgNativeTokenFee -- number
 -   *    schemeNativeTokenFee -- number		 +     *    schemeNativeTokenFee -- number

Does this comment is a valid one ?

Reply:
At this moment, yes, but as we add different schemes and voting machines, no

Reply:
Lets remove that if it is not valid.